packages/core/src/domain/ai-brief/generate.ts

```diff
@@ -64,3 +70,105 @@ export async function generateAiBriefBundleEntry(
 
   return undefined;
 }
+
+async function uploadAiBriefToS3(
+  aiBriefFhirResource: Binary,
+  cxId: string,
+  patientId: string,
+  log: typeof console.log
+) {
+  try {
+    const aiBriefWrappedInBundle = buildBundleFromResources({
+      resources: [aiBriefFhirResource],
+    });
+    const s3Utils = new S3Utils(Config.getAWSRegion());
+    const key = getAiBriefFileKey(cxId, patientId);
+    await s3Utils.uploadFile({
+      bucket: Config.getAiBriefBucketName(),
+      key,
+      file: Buffer.from(JSON.stringify(aiBriefWrappedInBundle)),
+    });
+  } catch (err) {
+    const msg = `Failed to upload AI Brief to S3`;
+    log(`${msg}: ${errorToString(err)}`);
+    capture.error(msg, {
+      extra: {
+        cxId,
+        patientId,
+        error: err,
+      },
+    });
+  }
+}
+
+/**
+ * Gets the AI Brief from S3, if it doesn't exist, generates a new one.
+ *
+ * @param cxId - The customer id.
+ * @param patientId - The patient id.
+ * @param bundle - The bundle to get the AI Brief from.
+ * @param log - The log function.
+ * @returns The AI Brief bundle entry.
+ */
+export async function getAiBriefFromS3({
+  cxId,
+  patientId,
+  bundle,
+  log,
+  aiBriefControls,
+}: {
+  cxId: string;
+  patientId: string;
+  bundle: Bundle<Resource>;
+  log: typeof console.log;
+  aiBriefControls?: AiBriefControls;
+}): Promise<BundleEntry<Binary> | undefined> {
+  log("Getting AI Brief from S3");
+  const s3Utils = new S3Utils(Config.getAWSRegion());
+  const filePath = getAiBriefFileKey(cxId, patientId);
+
+  try {
+    const aiBrief = await s3Utils.getFileContentsAsString(Config.getAiBriefBucketName(), filePath);
+
+    if (!aiBrief) {
+      log(`No AI Brief found in S3, generating new one`);
+      return await generateAiBriefBundleEntry(bundle, cxId, patientId, log, aiBriefControls);
+    }
+
+    log(`Got AI Brief from S3`);
+    const aiBriefBundle = parseFhirBundle(aiBrief) as Bundle<Binary>;
+    if (!aiBriefBundle) {
+      log(`Failed to parse AI Brief bundle, generating new one`);
+      return await generateAiBriefBundleEntry(bundle, cxId, patientId, log, aiBriefControls);
+    }
+
+    const aiBriefResource = getAiBriefResourceFromBundle(aiBriefBundle);
+    if (!aiBriefResource) {
+      log(`No AI Brief resource found in bundle, generating new one`);
+      return await generateAiBriefBundleEntry(bundle, cxId, patientId, log, aiBriefControls);
+    }
+
+    return buildBundleEntry(aiBriefResource);
+  } catch (err) {
+    const msg = `Failed to get AI Brief from S3`;
+    log(`${msg}: ${err}`);
+    log(`Generating new AI Brief due to error`);
+    return await generateAiBriefBundleEntry(bundle, cxId, patientId, log, aiBriefControls);
```

Comment:
once you refactor to use a params object, this can also be made a little cleaner by just having 

```generateAiBriefBundleEntry(params);```

instead